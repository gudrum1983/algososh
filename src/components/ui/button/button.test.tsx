import React from 'react';
import {Button} from './button';
import {fireEvent, render, screen} from "@testing-library/react";
const TestRenderer = require('react-test-renderer');

describe('test Button component', () => {

  it('should render an active button with text', () => {
    const id: string = "ActiveButtonWithText";
    const tree = TestRenderer
      .create(<Button
        data-testid={id}
        text={id}
        disabled={false}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render an active button without text', () => {
    const id = "ActiveButtonWithoutText";
    const tree = TestRenderer
      .create(<Button
        data-testid={id}
        disabled={false}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a button with a loading indicator', () => {
    const id = "ButtonWithLoadingIndicator";
    const tree = TestRenderer
      .create(<Button
        data-testid={id}
        text={id}
        isLoader={true}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a disabled button with text', () => {
    const id = "DisabledButton";
    const tree = TestRenderer
      .create(<Button
        data-testid={id}
        text={id}
        disabled={true}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('test callback Button component', () => {
  it('should invoke onClick button', () => {
    const id = "onClickActiveButton";
    const mockFn = jest.fn();
    render(<Button
      data-testid={id}
      onClick={mockFn}/>);
    const button = screen.getByTestId(id);
    fireEvent.click(button);
    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('should dont invoke onClick button with a loading indicator', () => {
    const id = "onClickButtonWithLoadingIndicator";
    const mockFn = jest.fn();
    render(<Button
      data-testid={id}
      onClick={mockFn}
      isLoader={true}/>);
    const button = screen.getByTestId(id);
    fireEvent.click(button);
    expect(mockFn).toHaveBeenCalledTimes(0);
  });

  it('should dont invoke onClick button disabled', () => {
    const id = "onClickButtonWithDisabled";
    const mockFn = jest.fn();
    render(<Button
      data-testid={id}
      onClick={mockFn}
      disabled={true}/>);
    const button = screen.getByTestId(id);
    fireEvent.click(button);
    expect(mockFn).toHaveBeenCalledTimes(0);
  });
});
