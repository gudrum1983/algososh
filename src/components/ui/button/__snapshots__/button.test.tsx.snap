// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button component should render a button with a loading indicator 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  data-testid="ButtonWithLoadingIndicator"
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`test Button component should render a disabled button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="DisabledButton"
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  >
    DisabledButton
  </p>
</button>
`;

exports[`test Button component should render an active button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="ActiveButtonWithText"
  disabled={false}
  type="button"
>
  <p
    className="text undefined"
  >
    ActiveButtonWithText
  </p>
</button>
`;

exports[`test Button component should render an active button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="ActiveButtonWithoutText"
  disabled={false}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
